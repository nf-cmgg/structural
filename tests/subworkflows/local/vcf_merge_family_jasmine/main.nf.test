nextflow_workflow {

    name "Test Workflow VCF_MERGE_FAMILY_JASMINE"
    script "subworkflows/local/vcf_merge_family_jasmine/main.nf"
    workflow "VCF_MERGE_FAMILY_JASMINE"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_merge_family_jasmine"

    test("homo_sapiens - one family, variant types") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"PosCon1", sample:"PosCon1", family:"fam", family_count:2, variant_type:"sv"],
                    file(params.sv_vcf1, checkIfExists:true)
                ],[
                    [id:"PosCon2", sample:"PosCon2", family:"fam", family_count:2, variant_type:"sv"],
                    file(params.sv_vcf2, checkIfExists:true)
                ],[
                    [id:"PosCon1", sample:"PosCon1", family:"fam", family_count:2, variant_type:"cnv"],
                    file(params.cnv_vcf1, checkIfExists:true)
                ],[
                    [id:"PosCon2", sample:"PosCon2", family:"fam", family_count:2, variant_type:"cnv"],
                    file(params.cnv_vcf2, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs
                ).match() }
            )
        }

    }

}
