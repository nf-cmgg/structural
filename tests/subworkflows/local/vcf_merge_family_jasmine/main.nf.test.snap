{
    "homo_sapiens - one family, no variant type": {
        "content": [
            [
                [
                    "fam.vcf.gz",
                    "fam.vcf.gz.tbi"
                ]
            ],
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:41:36.518885307"
    },
    "homo_sapiens - two samples, no variant type": {
        "content": [
            [
                [
                    "fam.vcf.gz",
                    "fam.vcf.gz.tbi"
                ]
            ],
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:41:51.906665609"
    },
    "homo_sapiens - two samples, variant types": {
        "content": [
            [
                [
                    "fam.cnv.vcf.gz",
                    "fam.cnv.vcf.gz.tbi"
                ],
                [
                    "fam.sv.vcf.gz",
                    "fam.sv.vcf.gz.tbi"
                ]
            ],
            10,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:40:56.296762018"
    },
    "homo_sapiens - one family, variant types": {
        "content": [
            [
                [
                    "fam.cnv.vcf.gz",
                    "fam.cnv.vcf.gz.tbi"
                ],
                [
                    "fam.sv.vcf.gz",
                    "fam.sv.vcf.gz.tbi"
                ]
            ],
            10,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:40:39.237020519"
    },
    "homo_sapiens - one family & one sample, variant types": {
        "content": [
            [
                [
                    "fam1.cnv.vcf.gz",
                    "fam1.cnv.vcf.gz.tbi"
                ],
                [
                    "fam1.sv.vcf.gz",
                    "fam1.sv.vcf.gz.tbi"
                ]
            ],
            10,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:41:20.367538619"
    },
    "homo_sapiens - one family & one sample, no variant types": {
        "content": [
            [
                [
                    "fam1.vcf.gz",
                    "fam1.vcf.gz.tbi"
                ]
            ],
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-12T11:42:07.437700938"
    }
}